### DISCUSSION TITLE
[Mac] Add scripting library

### DISCUSSION THREAD

**Discussion by Bakanych** [voxeltycoon/issues#436] [Modding]
Hi guys, 
there is a voice from the very bottom of the most famous gaming platform - macbook pro is talking.

It looks very tempting to create touch bar support for VT to represent game tools menu (industry, road, railroad, etc.) and maybe extend it in future with some nice fast action keys. 
I've tried some experiments with BTT preset already, but the only things I can do now is sending keyboard strokes. Even this simple approach needs some dirty hacks as it's not possible to switch between building modes using hotkey while one mode is active, e.g.: you cannot press 2 to select road mode while you're in railroad mode.

It will be nice to have mac scripting library for Voxel Tycoon to be able to interact with game UI via apple- or javascript automation.

Thanks and good luck with game launching!

**Comment by victorleblais** [comment]
If I'm not mistaken, the game will be opened up to modding using C# mods, since the game using that language (since it's using Unity).

But we don't have any idea of when it'll be implemented. On [the old trello roadmap](https://trello.com/c/d3L8ElXw/74-mod-api), it says it would be available when  the game release on Steam for it's alpha phase (out of itch pre-alpha).

**Comment by Bakanych** [comment]
As as first workaround it will be enough to be able to assign hotkey for every action in main game panel, e.g.: F1 - industry, ctrl-F2 - industry/warehouse, etc.

**Comment by Kpa6i** [comment]
you cannot press 2 to select road mode while you're in railroad mode.

 The same on Ubuntu. if I press 2 I'll go to road menu, then if press any other number it will open a sub menu from road. to select another menu need to press several times ESC, to go out of road menu and then press 3 (for example) to go to rails menu. 

I think the same on windows. So it's not only Mac issue

**Comment by Bakanych** [comment]
@Kpa6i Sure current hotkey system is not mac related. ESC-workaround is exactly how it's implemented in my current setup. The problem is, the more opened windows your have, the more times you need to press ESC. It's definitely not ideal hotkey system.

**Comment by Kpa6i** [comment]
yup. you are right. Will be better to add ability to set hot keys for different things

**Comment by victorleblais** [comment]
@Bakanych You should maybe switch the title to "Have keybinds on all actions and/or unique keybinds for most actions" or something like this since this issue is now about having keybinds to open/close all "build" menus.

Also, it makes it related to #32

**Comment by Bakanych** [comment]
@victorleblais It's not about keys only. It's way more reliable and powerful to interact with mac app using its script library rather than just send keystrokes.

**Comment by victorleblais** [comment]
@Bakanych I completely understand that what you're initially requesting is more than just keybinds, but as said previously, what you're asking will be possible when the modding API is available. In the meantime, I think having different keybind for all actions is still a useful feature that can be added to the game, and that could also help you whilst waiting for the modding tools to be available.
